app/__init__.py

```diff
@@ -281,6 +282,15 @@ def record_request_details():
         g.start = monotonic()
         g.endpoint = request.endpoint
 
+    @app.before_request
+    def check_content_length():
+        if (
+            request.content_length is not None
+            and current_app.config['MAX_CONTENT_LENGTH'] is not None
+            and request.content_length > current_app.config['MAX_CONTENT_LENGTH']
+        ):
+            raise WerkzeugRequestEntityTooLarge()
+
```

Comment:
calling `request.data` does the same thing, does it not? 
However, I don't object to this extra code because it is clear what it is doing.
As long as this doesn't affect the performance of our API I would leave this check here. I would be surprised if this does degrade performance.

Reply:
You're right, it does pretty much the same thing (if you follow the calls from here: https://github.com/pallets/werkzeug/blob/f3b081634c744ae1e0ae3f76034ee90bfdc2a808/src/werkzeug/wrappers/request.py#L447)

But I found it a bit too magic so, as you said, I tried to be more explicit about what's happening.

Reply:
`len(request.data)` will read the entire request data into memory in python (it'll probs still be in the socket at this point). `request.content_length` reads the CONTENT_LENGTH header.

Reply:
When were testing it yesterday, just by accessing `request.data` it was triggering the check and throwing the RequestEntityTooLarge, that's the "magic" bit I was referring to.